lightning/restore/restore.go

```diff
@@ -317,12 +322,14 @@ func (rc *RestoreController) restoreSchema(ctx context.Context) error {
 
 		for _, dbMeta := range rc.dbMetas {
 			task := log.With(zap.String("db", dbMeta.Name)).Begin(zap.InfoLevel, "restore table schema")
-
-			tablesSchema := make(map[string]string)
+			tablesSchemas := make([]tableSchema, 0, len(dbMeta.Tables))
```

Comment:
Please add a comment here why we don't use a map to hold these data, it can help others to understand it better

Reply:
Done, PTAL

Reply:
how looping over a map makes injection harder? you could make the failpoint return a table name.